### PR TITLE
typing: add check_untyped_defs

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
typing: add check_untyped_defs
    
Add the lint to check bodies of untyped definitions[1] which is enabled
by default in mypy v1.0.0[2].
   
Disable modules that do not pass this check with the intention to fix
them in the future but not add new modules or make the lint not pass for
already passing ones.
    
Rerefences:
[1] https://mypy.readthedocs.io/en/stable/config_file.html#confval-check_untyped_defs
[2] https://github.com/pandas-dev/pandas/commit/20e4c186191859dcde2437edacaee43d8d34dc46
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [x] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [x] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,8 +15,9 @@ skip = ["cloudinit/cmd/main.py", ".tox", "packages", "tools"]
 
 [tool.mypy]
 follow_imports = "silent"
-warn_unused_ignores = "true"
+check_untyped_defs = true
```
Comment (TheRealFalcon) [inline pyproject.toml:18]: Nit: Can we make types of the booleans consistent? `true` on this line but `"true"` on the following lines is a little weird.

**Review by TheRealFalcon — APPROVED** [review]
One inline nit, but otherwise LGTM!

**Review by a-dubs — APPROVED** [review]
Changes look good. Should we create an issue referencing this for future work needed to get this enabled on all modules?

File: pyproject.toml (hunk 1 above)
Comment (aciba90) [reply pyproject.toml:18]: Done.

**Pull request merged**